### ISSUE TITLE
Rule proposal: else, catch, finally, brace on same line

### ISSUE THREAD

**Issue by sveyret** [eslint/eslint#12559]
**Please describe what the rule should do:**

I am currently migrating from `tslint` to `eslint`. This proposal is to replace the `tslint` rule `one-line` (see https://palantir.github.io/tslint/rules/one-line/).

The purpose of the rule is to check that the specified token is in the same line as the preceding expression. Tokens which can be checked are:

- `catch`, which must be on same line as closing brace of `try`;
- `finally`, which must be on same line as closing brace of `catch`;
- `else`, which must be on same line as closing brace of `if`;
- open brace.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[x] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
if (a)
{ // <- Error: opening brace must be on same line as if
  try {
    // do something
  }
  catch (e) { // <- Error: catch must be on same line as closing brace
    throw e;
  }
}
else // <- Error: else must be on same line as closing brace
{ // <- Error: opening brace must be on same line as else
  // do something else
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

This seems to me that it is a rather basic code style rule, which is already required in many projects.

**Are you willing to submit a pull request to implement this rule?**

I may find time to create a PR, if such rule would be acceptable in the project.

**Comment by mdjermanovic** [comment]
Hi @sveyret, thanks for the new rule proposal!

Did you check the [brace-style](https://eslint.org/docs/rules/brace-style) rule, looks like the default `1tbs` option enforces this style.

**Comment by sveyret** [comment]
Thank you for your answer, @mdjermanovic. The brace-style seems to be what I was looking for. This issue can be closed.

**Comment by ilyavolodin** [comment]
Closing as the question has been answered.
